transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollStreamChannel.java

```diff
@@ -824,7 +824,7 @@ void epollInReady() {
                     allocHandle.incMessagesRead(1);
                     pipeline.fireChannelRead(byteBuf);
                     byteBuf = null;
-                } while (allocHandle.continueReading());
+                } while (config.isAutoRead() && allocHandle.continueReading());
```

Comment:
auto read, and other channel options, and controlling characteristics of the read loop belong to `RecvByteBufAllocator` implementations. For `EPOLL` there are a few special conditions that must be enforced and so `EpollRecvByteAllocatorHandle` exists to wrap any existing `RecvByteBufAllocator.Handle` to apply the special epoll conditions on top of the typical behavior.  

Lets look at implementations of `EpollRecvByteAllocatorHandle`:
1. `EpollRecvByteAllocatorMessageHandle` - The autoRead channel option will be enforced by the delegate `super.continueReading()` ... and if EDGE triggered we must always read until there is no more data ... so this is effectively a form of forced "auto read" while in the read loop.
2. `EpollRecvByteAllocatorStreamingHandle` - Again the autoRead channel option will be enforced by the delegate `super.continueReading()` but for EPOLL we continue reading as long as the last read operation actually read as much data as we requested ... if this is the case there more be more data to read. This is a slightly less strong form of "forcing auto read" but it seems to be doing the same thing.

So in summary I'm not sure why this additional call to `config.isAutoRead()` is necessary ... can you explain the scenario you are in (or create a unit tests) where this is necessary?